### PR TITLE
feat: add step to generate association data

### PR DESCRIPTION
**Pull request by vivienho**
## ✨ Context

We want to generate assocations from l2g evidence without relying on the platform etl.

## 🛠 What does this PR implement

This PR adds a step to generate direct and indirect associations from l2g evidence and saves them as parquet files.

## 🚦 Before submitting

- [x] Do these changes cover one single feature (one change at a time)?
- [x] Did you read the [contributor guideline](https://opentargets.github.io/gentropy/development/contributing/#contributing-checklist)?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you make sure there is no commented out code in this PR?
- [x] Did you follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standards in PR title and commit messages?
- [x] Did you make sure the branch is up-to-date with the `dev` branch?
- [x] Did you write any new necessary tests?
- [x] Did you make sure the changes pass local tests (`make test`)?
- [x] Did you make sure the changes pass pre-commit rules (e.g `poetry run pre-commit run --all-files`)?

### CODE REVIEW

File: src/gentropy/common/utils.py [hunk 1]
```diff
@@ -315,3 +315,22 @@ def copy_to_gcs(source_path: str, destination_blob: str) -> None:
...
+        Column: column of harmonic sums
+    """
+    return f.aggregate(
+        f.arrays_zip(
+            f.sort_array(input_array, False).alias("score"),
+            f.sequence(f.lit(1), f.size(input_array)).alias("pos")
+        ),
+        f.lit(0.0),
+        lambda acc, x: acc
+        + x["score"]/f.pow(x["pos"], 2)/f.lit(sum(1 / ((i + 1)**2) for i in range(100)))
```
Comment (addramir) [inline src/gentropy/common/utils.py:335]: Why only first 100 are used?
And It should be a devision by 1.644 somewhere...

Comment (vivienho) [reply src/gentropy/common/utils.py:335]: That part represents the division by ~1.644
I initially used first 1000 (1.6439..) but changed it to 100 (1.6349..)
Should I just change it to f.lit(1.644) ?

Comment (addramir) [reply src/gentropy/common/utils.py:335]: Sorry, misread it. All is fine. But I would use 1000 (to be consistent with the platform if documentation is correct)

**Review by DSuveges — APPROVED** [review]
This PR adds a new step to gentropy to generate disease/target association (direct and indirect) based on l2g evidence. This also requires the inclusion of a new method to compute harmonic sum of values.

**Pull request merged**